PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java

```diff
@@ -102,32 +104,70 @@ public WebApp executeSteps(@NotNull RunProcessHandler processHandler
         processHandler.setText(STOP_WEB_APP);
         webApp.stop();
 
+        processHandler.setText(GETTING_DEPLOYMENT_CREDENTIAL);
+        PublishingProfile profile = webApp.getPublishingProfile();
+
+        processHandler.setText(CONNECTING_FTP);
+        FTPClient ftp = WebAppUtils.getFtpConnection(profile);
+
         int indexOfDot = webAppSettingModel.getTargetName().lastIndexOf(".");
         String fileName = webAppSettingModel.getTargetName().substring(0, indexOfDot);
         String fileType = webAppSettingModel.getTargetName().substring(indexOfDot + 1);
 
-        switch (fileType) {
-            case MavenConstants.TYPE_WAR:
-                try (FileInputStream input = new FileInputStream(webAppSettingModel.getTargetPath())) {
-                    uploadWarArtifact(input, webApp, fileName, processHandler, telemetryMap);
-                }
-                break;
-            case MavenConstants.TYPE_JAR:
-                uploadJarArtifact(webAppSettingModel.getTargetPath(), webApp, processHandler, telemetryMap);
-                break;
-            default:
-                break;
+        int webConfigUploadCount = uploadWebConfigFile(ftp, webApp, fileType, processHandler);
+        telemetryMap.put("webConfigCount", String.valueOf(webConfigUploadCount));
+
+        try (FileInputStream input = new FileInputStream(webAppSettingModel.getTargetPath())) {
+            int artifactUploadCount = uploadArtifact(input, webApp, ftp, fileName, fileType, processHandler);
+            telemetryMap.put("artifactUploadCount", String.valueOf(artifactUploadCount));
         }
 
         processHandler.setText(START_WEB_APP);
         webApp.start();
 
+        processHandler.setText(LOGGING_OUT);
+        ftp.logout();
+        if (ftp.isConnected()) {
+            ftp.disconnect();
+        }
+
         String url = getUrl(webApp, fileName, fileType);
         processHandler.setText(DEPLOY_SUCCESSFUL);
         processHandler.setText("URL: " + url);
         return webApp;
     }
 
+    private int uploadArtifact(@NotNull FileInputStream input, @NotNull WebApp webApp, @NotNull FTPClient ftp,
+                               @NotNull String fileName, @NotNull String fileType,
+                               @NotNull RunProcessHandler processHandler)
+        throws IOException {
+        switch (fileType) {
+            case MavenConstants.TYPE_WAR:
+                // Workaround for Linux web apps, because unlike Windows ones, the webapps folder is not created in the
```

Comment:
Add the workaround for Linux Web Apps to ensure there are `webapps` folder to deploy to.